solver_options:
    regularization: 1e-3
    back_end: "qpoases"

stack:
    - ["Wheel1", "Wheel2", "Wheel3","Wheel4","ComXYZ"]
    - ["LeftArm", "RightArm", "PelvisRPY"]
    - ["Postural"]

constraints: ["JointLimits", "VelocityLimits"]

Collision:
    type: CollisionConstraint
    lib_name: libcartesio_collision_support.so
    bound_scaling: 0.1
    distance_threshold: 0.01
    max_pairs: 10
    collision_urdf_path: $(rospack find centauro_urdf)/urdf/capsule/centauro_capsules.urdf
    collision_srdf_path: $(rospack find centauro_srdf)/srdf/capsule/centauro_capsules.srdf

params:
 - &arm_indices [0, 1, 2]

PelvisRPY:
    type: "Cartesian"
    lambda: 0.1
    indices: [3, 4, 5]
    name: "PelvisRPY"
    distal_link: "pelvis"
    weight: 0.1
    active: false

ComXYZ:
    type: "Com"
    lambda: 0.1
    indices: [0, 1, 2]
    weight: 10
    active: true

Wheel1:
    type: "Cartesian"
    name: "FL_wheel"
    distal_link: "wheel_1"
    lambda: 0.1

Wheel2:
    type: "Cartesian"
    name: "FR_wheel"
    distal_link: "wheel_2"
    lambda: 0.1

Wheel3:
    type: "Cartesian"
    name: "HL_wheel"
    distal_link: "wheel_3"
    lambda: 0.1

Wheel4:
    type: "Cartesian"
    name: "HR_wheel"
    distal_link: "wheel_4"
    lambda: 0.1

LeftArm:
    type: "Cartesian"
    name: "left_hand"
    distal_link: "arm1_8"
    lambda: 0.1
    indices: *arm_indices
#    base_link: "pelvis"

RightArm:
    type: "Cartesian"
    name: "right_hand"
    distal_link: "arm2_8"
    lambda: 0.1
    indices: *arm_indices
    #    base_link: "pelvis"
    
Postural:
    type: "Postural"
    lambda: 0.01
    weight:
        VIRTUALJOINT_1: 0.0
        VIRTUALJOINT_2: 0.0
        VIRTUALJOINT_3: 0.0
        VIRTUALJOINT_4: 0.0
        VIRTUALJOINT_5: 0.0
        VIRTUALJOINT_6: 0.0

JointLimits:
    type: "JointLimits"

VelocityLimits:
    type: "VelocityLimits"
